{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Demo\n",
    "\n",
    "> comming soon..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting git+https://github.com/nicomarr/scholaris.git\n",
      "  Cloning https://github.com/nicomarr/scholaris.git to /private/var/folders/fm/9x31wj1d4yn0yk43g3qwzn_00000gq/T/pip-req-build-tfg6foag\n",
      "  Running command git clone --filter=blob:none --quiet https://github.com/nicomarr/scholaris.git /private/var/folders/fm/9x31wj1d4yn0yk43g3qwzn_00000gq/T/pip-req-build-tfg6foag\n",
      "  Resolved https://github.com/nicomarr/scholaris.git to commit ed279e1c7287b82b48f2f1d9445f1619eae9d6f3\n",
      "  Installing build dependencies ... \u001b[?25ldone\n",
      "\u001b[?25h  Getting requirements to build wheel ... \u001b[?25ldone\n",
      "\u001b[?25h  Preparing metadata (pyproject.toml) ... \u001b[?25ldone\n",
      "\u001b[?25hNote: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "%pip install git+https://github.com/nicomarr/scholaris.git"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#| hide\n",
    "import ollama\n",
    "from scholaris.core import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded Semantic Scholar API key from the environment variables.\n",
      "Loaded email address from the environment variables.\n",
      "A local directory ../data already exists for storing data files. No of files: 1\n",
      "\n"
     ]
    }
   ],
   "source": [
    "assistant = Assistant()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Selecting tools...\n",
      "\n",
      "[{'function': {'name': 'get_file_names', 'arguments': {'ext': 'pdf'}}}]\n",
      "Calling get_file_names() with arguments {'ext': 'pdf'}...\n",
      "\n",
      "Function response:\n",
      "List of file names with the specified extensions in the local data directory: jci.insight.144499.v2.pdf\n",
      "\n",
      "Generating final response...\n",
      "\n",
      "\n",
      "\u001b[94mI\u001b[0m\u001b[94m have\u001b[0m\u001b[94m access\u001b[0m\u001b[94m to\u001b[0m\u001b[94m the\u001b[0m\u001b[94m following\u001b[0m\u001b[94m local\u001b[0m\u001b[94m PDF\u001b[0m\u001b[94m files\u001b[0m\u001b[94m:\n",
      "\n",
      "\u001b[0m\u001b[94m1\u001b[0m\u001b[94m.\u001b[0m\u001b[94m j\u001b[0m\u001b[94mci\u001b[0m\u001b[94m.ins\u001b[0m\u001b[94might\u001b[0m\u001b[94m.\u001b[0m\u001b[94m144\u001b[0m\u001b[94m499\u001b[0m\u001b[94m.v\u001b[0m\u001b[94m2\u001b[0m\u001b[94m.pdf\u001b[0m\u001b[94m\u001b[0m"
     ]
    }
   ],
   "source": [
    "#| hide\n",
    "assistant.chat(\"Which local PDF files do you have access to?\", show_progress=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Selecting tools...\n",
      "\n",
      "[{'function': {'name': 'get_titles_and_first_authors', 'arguments': {}}}]\n",
      "Calling get_titles_and_first_authors() with arguments {}...\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Extracting titles and first authors: 100%|██████████| 1/1 [00:08<00:00,  8.73s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Function response:\n",
      "[\n",
      "  {\n",
      "    \"title\": \"Distinct antibody repertoires against endemic human coronaviruses in children and adults\",\n",
      "    \"first_author\": \"Taushif Khan\",\n",
      "    \"file_name\": \"jci.insight.144499.v2.pdf\"\n",
      "  }\n",
      "]\n",
      "\n",
      "Generating final response...\n",
      "\n",
      "\n",
      "\u001b[94mThe\u001b[0m\u001b[94m document\u001b[0m\u001b[94m has\u001b[0m\u001b[94m a\u001b[0m\u001b[94m title\u001b[0m\u001b[94m:\u001b[0m\u001b[94m \"\u001b[0m\u001b[94mDistinct\u001b[0m\u001b[94m antibody\u001b[0m\u001b[94m rep\u001b[0m\u001b[94merto\u001b[0m\u001b[94mires\u001b[0m\u001b[94m against\u001b[0m\u001b[94m endemic\u001b[0m\u001b[94m human\u001b[0m\u001b[94m coron\u001b[0m\u001b[94mavir\u001b[0m\u001b[94muses\u001b[0m\u001b[94m in\u001b[0m\u001b[94m children\u001b[0m\u001b[94m and\u001b[0m\u001b[94m adults\u001b[0m\u001b[94m\".\u001b[0m\u001b[94m\u001b[0m"
     ]
    }
   ],
   "source": [
    "#| hide\n",
    "assistant.chat(\"Does this document have a title?\", show_progress=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Selecting tools...\n",
      "\n",
      "[{'function': {'name': 'extract_text_from_pdf', 'arguments': {'file_name': 'jci.insight.144499.v2.pdf'}}}]\n",
      "Calling extract_text_from_pdf() with arguments {'file_name': 'jci.insight.144499.v2.pdf'}...\n",
      "\n",
      "Function response:\n",
      "page 0 of 15\n",
      "1\n",
      "RESEARCH ARTICLEConflict of interest: The authors have \n",
      "declared that no conflict of interest \n",
      "exists.\n",
      "Copyright: © 2021, Khan et al. This is \n",
      "an open access article published under \n",
      "the terms of the Creative Commons \n",
      "Attribution 4.0 International License.\n",
      "Submitted: September 21, 2020 \n",
      "Accepted: January 13, 2021 \n",
      "Published: January 26, 2021\n",
      "Reference information: JCI Insight. \n",
      "2021;6(4):e144499. \n",
      "https://doi.org/10.1172/jci.\n",
      "insight.144499.Distinct antibody repertoires against \n",
      "endemic human coronaviruses in  \n",
      "children and adults\n",
      "Taushif Khan,1 Mahbuba Rahman,1 Fatima Al Ali,1 Susie S. Y . Huang,1 Manar Ata,1 Qian Zhang,2  \n",
      "Paul Bastard,2,3,4 Zhiyong Liu,2 Emmanuelle Jouanguy,2,3,4 Vivien Béziat,2,3,4 Aurélie Cobat,2,3,4  \n",
      "Gheyath K. Nasrallah,5,6 Hadi M. Yassine,5,6 Maria K. Smatti,6 Amira Saeed,7 Isabelle Vandernoot,8 \n",
      " Jean-Christophe Goffard,9 Guillaume Smits,8 Isabelle Migeotte,10 Filomeen Haerynck,11 Isabelle Meyts,12,13 \n",
      "Laurent Abel,2,3,4 Jean-Laurent Casanova,2,3,4,14 Mohammad R. Hasan,7,15 and Nico Marr1,16\n",
      "1Research Branch, Sidra Medicine, Doha, Qatar. 2St. Giles Laboratory of Human Genetics of Infectious Diseases, Rockefeller \n",
      "Branch, Rockefeller University, New York, New York, USA. 3Laboratory of Human Genetics of Infectious Diseases, Necker \n",
      "Branch, INSERM U1163, Necker Hospital for Sick Children, Paris, France. 4University of Paris, Imagine Institute, Paris, \n",
      "France. 5College of Health Sciences, QU Health, Qatar University, Doha, Qatar. 6Biomedical Research Center, Qatar \n",
      "University, Doha, Qatar. 7Department of Pathology, Sidra Medicine, Doha, Qatar. 8Center of Human Genetics, 9Department \n",
      "of Internal Medicine, and 10Fonds de la Recherche Scientifique (FNRS) and Center of Human Genetics, Hôpital Erasme, \n",
      "Université Libre de Bruxelles, Brussels, Belgium. 11Department of Pediatric Pulmonology and Immunology, Department \n",
      "of Pediatrics and Internal Medicine, Center for Primary Immunodeficiencies Ghent, Jeffrey Modell Foundation Diagnostic \n",
      "and Research Center, Ghent University Hospital, Belgium. 12Laboratory for Inborn Errors of Immunity, Department of \n",
      "Microbiology, Immunology and Transplantation, and Department of Pediatrics, University Hospitals Leuven, KU Leuven, \n",
      "Belgium. 13Department of Pediatrics, University Hospitals Leuven, KU Leuven, Belgium. 14Howard Hughes Medical \n",
      "Institute, New York, New York, USA. 15Weill Cornell Medical College in Qatar, Doha, Qatar. 16College of Health and Life \n",
      "Sciences, Hamad Bin Khalifa University, Doha, Qatar.\n",
      "Introduction\n",
      "Four endemic human-tropic coronaviruses (HCoVs) are commonly associated with respiratory illness in humans, \n",
      "namely HCoV-229E, -NL63, -OC43, and -HKU1 (1–4). Clinical outcomes of acute infection with these HCoVs \n",
      "range from mild upper respiratory tract infections in most patients, to viral bronchiolitis and pneumonia more \n",
      "rarely in patients, the latter requiring hospitalization (5). The ratio of more severe versus mild outcomes of acute \n",
      "infection with endemic HCoVs is largely comparable to that of other “common cold” viruses, such as human \n",
      "respiratory syncytial virus (HRSV), human rhinoviruses (HRVs), human adenoviruses, and human parainfluen-\n",
      "za viruses, albeit with differences in seasonality and prevalence of the viruses depending on the species (5–7). In Four endemic human coronaviruses (HCoVs) are commonly associated with acute respiratory \n",
      "infection in humans. B cell responses to these “common cold” viruses remain incompletely \n",
      "understood. Here we report a comprehensive analysis of CoV-specific antibody repertoires in \n",
      "231 children and 1168 adults using phage immunoprecipitation sequencing. Seroprevalence of \n",
      "antibodies against endemic HCoVs ranged between approximately 4% and 27% depending on the \n",
      "species and cohort. We identified at least 136 novel linear B cell epitopes. Antibody repertoires \n",
      "against endemic HCoVs were qualitatively different between children and adults in that anti-\n",
      "HCoV IgG specificities more frequently found among children targeted functionally important \n",
      "and structurally conserved regions of the spike, nucleocapsid, and matrix proteins. Moreover, \n",
      "antibody specificities targeting the highly conserved fusion peptide region and S2′  cleavage site \n",
      "of the spike protein were broadly cross-reactive with peptides of epidemic human and nonhuman \n",
      "coronaviruses. In contrast, an acidic tandem repeat in the N-terminal region of the Nsp3 subdomain \n",
      "of the HCoV-HKU1 polyprotein was the predominant target of antibody responses in adult \n",
      "donors. Our findings shed light on the dominant species-specific and pan-CoV target sites of \n",
      "human antibody responses to coronavirus infection, thereby providing important insights for the \n",
      "development of prophylactic or therapeutic monoclonal antibodies and vaccine design.page 1 of 15\n",
      "2\n",
      "RESEARCH ARTICLE JCI Insight 2021;6(4):e144499  https://doi.org/10.1172/jci.insight.144499addition to the 4 endemic HCoVs, 3 human-tropic epidemic coronaviruses (CoVs) have emerged over the last 2 \n",
      "decades, namely severe acute respiratory syndrome–CoV (SARS-CoV) (8), Middle East respiratory syndrome–\n",
      "CoV (MERS-CoV) (9), and SARS-CoV-2 (10), the etiological agent of coronavirus disease 2019 (COVID-19), \n",
      "which has now reached pandemic proportions (11). Similar to endemic HCoVs, infection of humans with epi-\n",
      "demic CoVs is associated with a wide range of outcomes but leads more frequently to severe clinical manifesta-\n",
      "tions, such as acute respiratory distress syndrome (ARDS) (12–14). Phylogenetic analyses suggest that, similar \n",
      "to these epidemic CoVs, all endemic HCoVs are of zoonotic origin, and their possible ancestors share similar \n",
      "natural animal reservoirs and intermediate hosts (6). HCoV-229E may have been transferred from dromedary \n",
      "camels, similar to MERS-CoV , while HCoV-OC43 is thought to have emerged more recently from ancestors \n",
      "in domestic animals such as cattle or swine in the context of a pandemic at the end of the 19th century (6, 15).\n",
      "The wide variability in transmissibility and clinical manifestations of infections by endemic and epidemic \n",
      "CoVs among humans remains poorly understood. On the population level, the case fatality rate is highest for \n",
      "MERS (approximately 34%– 37%), and several risk factors are associated with progression to ARDS in MERS, \n",
      "SARS, and COVID-19 cases, including old age (i.e., people aged 65 years or over), diabetes mellitus, hyperten-\n",
      "sion, cancer, renal and lung disease, and coinfections (12, 16). Nonetheless, even MERS-CoV infection among \n",
      "humans can run a completely asymptomatic course in some cases, particularly among children (17–19). There \n",
      "is evidence that children are generally less susceptible to infection with epidemic CoVs, and once infected, they \n",
      "are less likely to experience severe outcomes compared with adults, although this important association and the \n",
      "underlying reasons remain poorly understood (12, 18, 20, 21). Importantly, it remains unclear to what extent \n",
      "preexisting immunity from past infections with endemic HCoVs provides some degree of cross-protection and \n",
      "affects clinical outcomes of infection with the epidemic SARS-CoV-2 or MERS-CoV . Our overall understand-\n",
      "ing of the immunity induced by natural infection with endemic HCoVs remains very limited. Serological stud-\n",
      "ies have shown some degree of cross-reactive antibodies in patients with past CoV infections, but many of these \n",
      "studies were limited in sample size and often focused on specific viral antigens only (22–25). Depending on \n",
      "their binding affinity and specificities, such cross-reactive antibodies could have no effect on clinical outcomes, \n",
      "may provide protection from severe disease to some degree, or may lead to antibody-dependent enhancement \n",
      "of disease — the latter can be a major obstacle in vaccine development (26). Interestingly, 2 recent studies from \n",
      "independent groups have shown that a considerable proportion of individuals without a history of SARS-\n",
      "CoV-2 infection have SARS-CoV-2–reactive T cells, which suggests that cross-reactive T cell subsets originating \n",
      "from past infections by endemic HCoVs may play a role in the clinical course of infection with the phylogeneti-\n",
      "cally related epidemic CoVs (27, 28). A systematic assessment to elucidate the immunodominant B cell antigen \n",
      "determinants of endemic HCoVs has not been done. We hypothesized that a fraction of the general population \n",
      "also have antibodies generated during past encounters with “common cold” coronaviruses that cross-react with \n",
      "proteins of epidemic CoVs. This may affect the dynamics of sporadic MERS outbreaks that mostly occur in \n",
      "the Middle East and the current COVID-19 pandemic.\n",
      "Results\n",
      "To gain a deeper insight into human antibody responses to endemic HCoVs, we performed phage immuno-\n",
      "precipitation sequencing (PhIP-Seq) (29, 30) on previously collected serum or plasma samples obtained from \n",
      "1431 human subjects from 3 cohorts. These included (a) healthy male adult blood donors (ABD) with diverse \n",
      "ethnic backgrounds and nationalities (Supplemental Figure 1A; supplemental material available online with this \n",
      "article; https://doi.org/10.1172/jci.insight.144499DS1); (b) adult male and female participants of a national \n",
      "cohort study — the Qatar Biobank (QBB) (31) — representing the general population (Supplemental Figure \n",
      "1B); and (c) pediatric outpatients and inpatients who were tested for metabolic conditions unrelated to infection, \n",
      "chronic disease, or cancer (Methods and Supplemental Figure 1C). The samples were collected prior to the cur-\n",
      "rent COVID-19 outbreak (Methods). In brief, PhIP-Seq allowed us to obtain comprehensive antiviral antibody \n",
      "repertoires across individuals in our 3 human cohorts using phage display of oligonucleotide-encoded pepti-\n",
      "domes, followed by immunoprecipitation and massive parallel sequencing (29, 30). The VirScan phage library \n",
      "used for PhIP-Seq in the present study comprised peptides derived from viral proteins — each represented by \n",
      "peptide tiles of up to 56 amino acids in length that overlap by 28 amino acids — which collectively encompass \n",
      "the proteomes of a large number of viral species, including HCoV-229E, -NL63, -HKU1, and -OC43 (29, 30). \n",
      "Proteins of endemic HCoVs that were represented in the VirScan phage library included the ORF1ab replicase \n",
      "polyprotein (pp1ab), the spike glycoprotein (S), the matrix glycoprotein (M), the nucleocapsid protein (N), and \n",
      "gene products of the species- and strain-specific open reading frames (ORFs) encoded in the 3′  region of the page 2 of 15\n",
      "3\n",
      "RESEARCH ARTICLE JCI Insight 2021;6(4):e144499  https://doi.org/10.1172/jci.insight.144499viral genomes (Supplemental Table 1). Of note, we utilized an expanded version of the VirScan phage library \n",
      "(32, 33), which also encompassed peptides from a number of proteins of human epidemic and nonhuman CoV \n",
      "isolates, including MERS-CoV , SARS-CoV , as well as bat, bovine, porcine, and feline isolates belonging to the \n",
      "alpha- and beta-CoV genera, albeit with varying coverage of the viral peptidomes owing to the limitation in \n",
      "available sequence data for the latter isolates in UniProt (Supplemental Table 1). SARS-CoV-2 peptides were not \n",
      "included in the VirScan phage library used in our study.\n",
      "We were able to obtain antibody repertoires for 1399 individuals from the human cohorts described above \n",
      "(Supplemental Table 2). Using stringent filter criteria (Methods), we identified a total of 417 out of 2498 pep-\n",
      "tides and potential antigens from endemic HCoVs with our screen that were significantly enriched in at least \n",
      "3 of all 1399 analyzed individuals. A total of 103 peptides from endemic HCoVs were enriched in at least 1% \n",
      "of the samples and therefore considered to contain potentially immunodominant regions (Supplemental Table \n",
      "3). Only 33 of the 417 peptides enriched in at least 3 samples shared linear sequence homology with epitopes \n",
      "that have previously been reported (34) (Supplemental Figure 2). To estimate number of newly identified linear \n",
      "B cell epitopes, we assigned each CoV-derived peptide to clusters of peptides that share at least 7 amino acids \n",
      "linear sequence identity — the estimated size of a linear B cell epitope (Methods). The enriched peptides could \n",
      "be assigned to 149 clusters for which at least 2 peptides shared linear sequence identity of at least 7 amino acids \n",
      "(Supplemental Tables 3 and 4). Only 13 clusters also shared at least 7 amino acids linear sequence identity with \n",
      "known linear B cell epitopes. Consequently, we have identified a minimum of 136 new linear epitopes, including \n",
      "25 new immunodominant linear B cell epitopes — i.e., B cell epitopes targeted in at least 1% of all individuals \n",
      "and not already reported in the Immune Epitope Database: https://www .iedb.org (34) (Supplemental Table 3).\n",
      "Next we assessed the seroprevalence of HCoV-229E, -NL63, -HKU1, and -OC43 in the 3 cohorts sepa-\n",
      "rately. To do so, we computed species score values as described earlier (30, 32, 35) by counting the significantly \n",
      "enriched peptides for a given HCoV species that shared less than 7 amino acids linear sequence identity. We \n",
      "considered an individual seropositive for any of the endemic HCoVs if the number of nonhomologous peptides \n",
      "enriched in a given sample met our previously established species-specific cutoff value (Methods). Seroprev-\n",
      "alence for endemic HCoVs ranged from approximately 4% to approximately 27%, depending on the species \n",
      "and cohort (Figure 1A), and also varied when stratifying the subjects by age group or sex (Supplemental Table \n",
      "5). Interestingly, we found a marginal but significant negative association between age and seroprevalence of \n",
      "HCoV-OC43 (β  = –0.175) and -NL63 (β  = –0.315), as well as a marginally positive association between male sex \n",
      "and seroprevalence for any of the endemic HCoVs (β  ≤ 0.2) (Figure 1B). The species score values (i.e., the anti-\n",
      "body repertoire breadth for each HCoV species) did not differ substantially between seropositive individuals of \n",
      "our 3 cohorts (Supplemental Figure 3). However, principal component analysis revealed considerable qualitative \n",
      "differences in the antibody repertoires between our cohorts and in particular between pediatric and adult subjects \n",
      "(Figure 2A). For comparison, we also performed the same analysis on enriched peptides from other common \n",
      "respiratory viruses, including HRSV , HRV A, HRV B, and influenza B virus. As expected, seroprevalence was \n",
      "Figure 1. Seroprevalence of endemic HCoVs. ( A) Dot plot depicting the seroprevalence of the 4 endemic HCoVs among subjects included in the downstream  \n",
      "analysis (n  = 1399) after stratification by cohort. Gray bars depict the mean seroprevalence value for each species; error bars depict the SD. QBB, Qatar Biobank \n",
      "cohort; ABD, adult (male) blood bank donors; PED, pediatric study subjects. (B ) Coefficient of association (β ) with 95% CI of seroprevalence for each HCoV with \n",
      "male sex (blue), female sex (pink), or age (black). Only features that had a P  value of association less than or equal to 0.001 are shown.page 3 of 15\n",
      "4\n",
      "RESEARCH ARTICLE JCI Insight 2021;6(4):e144499  https://doi.org/10.1172/jci.insight.144499considerably higher (68% to 99%) for HRSV , HRV A, and HRV B and somewhat higher (29% to 47%) for influ-\n",
      "enza B virus (Supplemental Figure 4A). However, contrary to antiviral antibody responses to endemic HCoVs, \n",
      "we did not find considerable variance in the antibody repertoires to other respiratory viruses when comparing \n",
      "age groups and cohorts (Supplemental Figure 4B). We also analyzed the enriched antigenic peptides for each \n",
      "endemic HCoV species separately and found that most variance in the antibody repertoires between cohorts \n",
      "and age groups was attributable to past infections with HCoV-HKU1 and -229E (Supplemental Figure 4C). To \n",
      "determine the antibody specificities responsible for most of the variance in the antiviral response to endemic \n",
      "HCoVs between adults and children (i.e., to identify those peptides that were significantly more or less frequently \n",
      "enriched when comparing adult and pediatric donors), we applied Fisher’s exact test and computed log odds \n",
      "ratios (lod) for each of the significantly enriched peptides. We found that antibody specificities in samples of \n",
      "pediatric study subjects predominantly targeted different antigenic regions in the S protein (mean lod = 3.35, \n",
      "SD = 2.12) and the N protein (mean lod = 2.21, SD = 1.41) and diverse antigenic sites in pp1ab, whereas pep-\n",
      "tides encoding a single linear B cell epitope of pp1ab (cluster 22) appeared to be the predominant target of IgG \n",
      "antibodies among adult donors (mean lod = –4.7, SD = 1.16) (Figure 2B, Table 1, and Supplemental Table 6).\n",
      "Intriguingly, multiple sequence alignments of frequently enriched peptides with the full-length pro-\n",
      "teins of various CoVs revealed that antibody specificities predominantly found in pediatric study subjects \n",
      "targeted immunodominant epitopes that encode functionally important and highly conserved regions \n",
      "of the structural proteins. These included regions in the S1 subunit of the S protein that are important \n",
      "for receptor binding (36–39), as well as the regions resembling the proteolytic cleavage sites and fusion \n",
      "peptide of the S2 subunit (Figure 3 and Supplemental Figure 6). Of note, the immunodominant region \n",
      "spanning the furin-like S2′  cleavage site in the S2 subunit resembled one of the most conserved regions \n",
      "of the S protein, both in amino acid sequence (R¯SA[I/L]ED[I/L]LF) (Figure 3E) and in protein struc-\n",
      "ture because it formed an accessible α -helix within the fusion peptide region (Supplemental Figure 6) \n",
      "(40). Moreover, we identified potential antibody binding sites in the N-terminal RNA binding domain, \n",
      "serine/arginine-rich region, and C-terminal dimerization domain of the N protein (Figure 4, A and B). \n",
      "Although the predicted antibody binding sites in the N-terminal RNA binding domain and the C-ter-\n",
      "minal dimerization domain of the N protein appeared to be less conserved between different species in \n",
      "the primary amino acid sequence (Figure 4, C and D), both domains were structurally conserved in the \n",
      "regions that we found to be immunodominant (Supplemental Figure 7). We also found that antibodies  \n",
      "in children more frequently targeted the C-terminal domain of the M protein (Supplemental Figure 5C and \n",
      "Table 1) and the small accessory ORF8 protein (also known as N2) of HCoV-HKU1 (Table 1). Although \n",
      "ORF8 and N share the same coding sequence in the viral RNA genome, the reading frame is different and \n",
      "the amino acid sequences are not homologous. On the contrary, antibody specificities predominantly found \n",
      "in adults primarily targeted a region of the pp1ab that is specific to HCoV-HKU1 and contains an acidic \n",
      "Figure 2. Qualitative differences in antibody repertoires among cohorts and age groups. ( A) Principal component analysis of 417 peptides from endemic HCoVs \n",
      "that were found to be enriched in at least 3 samples. QBB (n  = 798); ABD (n  = 370); PED (n  = 231). (B ) Differential enrichment analysis to determine the peptides \n",
      "that are either more or less frequently enriched in children versus adults (including subjects of both adult cohorts, namely QBB and ABD). We considered a peptide \n",
      "significantly more or less frequently enriched among children if the OR was ≥  2 or ≤  –2, respectively, and the P  value was less than or equal to 0.005 (Fisher’s exact \n",
      "test). pp1ab, ORF1ab replicase polyprotein; S, spike glycoprotein; M, matrix glycoprotein; N, nucleocapsid protein; ORF8, open reading frame 8 protein.page 4 of 15\n",
      "5\n",
      "RESEARCH ARTICLE JCI Insight 2021;6(4):e144499  https://doi.org/10.1172/jci.insight.144499tandem repeat (NDDE[D/H]VVTGD), which is located upstream of the papain-like protease 1 domain \n",
      "(Supplemental Figure 5D and Supplemental Table 6).\n",
      "Given the high degree of sequence conservation among some of the immunodominant regions in pro-\n",
      "teins of endemic HCoVs we have identified, we also explored the extent to which antibody specificities to \n",
      "these regions cross-react with peptides from epidemic CoVs and nonhuman CoV isolates. For this purpose, \n",
      "we assessed the enrichment of peptides derived from SARS-CoV , MERS-CoV , as well as bovine, porcine, \n",
      "bat, and feline CoV isolates (Supplemental Table 1), applying the same approach and stringent filter criteria \n",
      "as described above for peptides of endemic HCoVs. Indeed, we identified several S protein– and N protein–\n",
      "derived peptides from epidemic CoVs or nonhuman isolates that were significantly enriched in our PhIP-Seq \n",
      "assay and that shared sequence similarity with peptides from HCoVs (Figure 5). As expected based on the \n",
      "results from multiple sequence alignments described above, antibody specificities targeting the highly con-\n",
      "served amino acid motif (RSA[I/L]ED[I/L]LF) spanning the furin-like S2′  cleavage site of the S protein were \n",
      "broadly cross-reactive to several orthologous peptides from MERS-CoV , SARS-CoV , and nonhuman CoV \n",
      "isolates (Figure 5A). Cross-reactivity of antibodies targeting other functionally important but less conserved Table 1. List of peptides that were more frequently enriched among children (n = 231) in comparison with adults (n = 1168)\n",
      "UniProtKB entry StartAEndAProtein symbol Species lod Log10 P value Cluster no. Cluster annotationB/peptide location\n",
      "P0C6X6 1401 1456 pp1ab HCoV-OC43 6.4 4.7 136 Papain-like proteinase\n",
      "Q0ZJJ1 813 868 pp1ab HCoV-HKU1 6.3 3.9 102 Papain-like proteinase\n",
      "Q0ZJJ1 225 280 pp1ab HCoV-HKU1 6.0 3.1 38 3C-like proteinase\n",
      "P0C6X2 7029 7084 pp1ab HCoV-HKU1 6.0 3.1 117 2′-O-methyltransferase\n",
      "Q5MQD0 365 420 S HCoV-HKU1 6.0 3.1 189 S1 chain region\n",
      "Q0ZJJ1 5825 5880 pp1ab HCoV-HKU1 5.7 2.4 9 RNA virus helicase\n",
      "P0C6X5 3753 3808 pp1ab HCoV-NL63 5.7 2.4 10 Non-structural protein 8\n",
      "Q0QJI4 337 392 S HCoV-OC43 5.7 2.4 98 S1/S2 cleavage site\n",
      "P0C6X3 5489 5544 pp1ab HCoV-HKU1 4.5 12.3 8 Helicase\n",
      "Q6Q1R9 197 226 M HCoV-NL63 3.2 3.2 185 Interaction with N protein region\n",
      "P0C6X6 2381 2436 pp1ab HCoV-OC43 3.2 3.2 253 Non-structural protein 3\n",
      "Q6Q1S2 421 476 S HCoV-NL63 3.0 2.5 131 S1 chain region\n",
      "Q6Q1S2 449 504 S HCoV-NL63 3.0 2.5 131 S1 chain region\n",
      "P0C6X1 1373 1428 pp1ab HCoV-229E 3.0 2.5 235 Non-structural protein 3\n",
      "P0C6X2 3305 3360 pp1ab HCoV-HKU1 2.7 5.0 13 Region of 3C-like proteinase\n",
      "Q6Q1R8 29 84 N HCoV-NL63 2.5 6.0 49 N-terminal RNA binding domain\n",
      "E2DNV6 1 56 N HCoV-NL63 2.2 7.3 37 N-terminal RNA binding domain\n",
      "Q6Q1R8 169 224 N HCoV-NL63 1.8 2.4 37 N-terminal RNA binding domain\n",
      "Q5SBN5 1 56 N HCoV-NL63 1.8 2.4 49 N-terminal RNA binding domain\n",
      "Q6Q1R8 197 252 N HCoV-NL63 1.8 7.8 37 N-terminal RNA binding domain\n",
      "Q5MQC5 1 56 ORF8 HCoV-HKU1 1.7 5.9 41 N-terminal RNA binding domain\n",
      "Q6Q1R8 337 377 N HCoV-NL63 1.6 24.6 132 C-terminal of dimerization domain\n",
      "P0C6X2 5461 5516 pp1ab HCoV-HKU1 1.5 2.8 34 RNA virus helicase C-terminal\n",
      "P0C6X5 4285 4340 pp1ab HCoV-NL63 1.5 2.5 78 RNA-directed RNA polymerase\n",
      "E2DNV6 29 84 N HCoV-NL63 1.5 8.5 37 N-terminal RNA binding domain\n",
      "Q5SBN5 29 83 N HCoV-NL63 1.5 3.2 49 N-terminal RNA binding domain\n",
      "Q0ZJJ1 1121 1176 pp1ab HCoV-HKU1 1.4 4.6 12 Asp-rich region, Papain-like proteinase\n",
      "P15423 645 700 S HCoV-229E 1.3 2.8 16 Furin-like S2′ cleavage site\n",
      "G9G2X4 1 56 N HCoV-HKU1 1.2 5.3 41 N-terminal RNA binding domain\n",
      "Q01455 197 230 M HCoV-OC43 1.2 3.5 147 Interaction with N protein region\n",
      "Q6Q1R8 309 364 N HCoV-NL63 1.1 9.5 132 C-terminus of dimerization domain\n",
      "P0C6X2 4509 4564 pp1ab HCoV-HKU1 1.1 3.6 2 RNA-directed RNA polymerase\n",
      "Q0QJI4 757 812 S HCoV-OC43 1.0 4.9 46 Receptor binding domain\n",
      "Q5MQC5 29 84 ORF8 HCoV-HKU1 1.0 3.0 41 N-terminal RNA binding domain\n",
      "For a full list of differentially enriched peptides, see Supplemental Table 6. Only peptides with a log odds ratio (lod) ≥ 1 and a P < 0.005 (Fisher’s exact \n",
      "test) are listed. Immunodominant peptides (i.e., peptides found to be significantly enriched in ≥1% of samples of the tested individuals [n = 1399]) are \n",
      "marked in bold font. AStart and end position of enriched peptides relative to the amino acid sequences in UniProtKB. BCluster annotation was adapted from \n",
      "UniProtKB entry descriptions and features.page 5 of 15\n",
      "6\n",
      "RESEARCH ARTICLE JCI Insight 2021;6(4):e144499  https://doi.org/10.1172/jci.insight.144499\n",
      "Figure 3. Antigenic regions and predicted antibody binding sites of the S protein. ( A) Schematic representation of the S protein of SARS-CoV (UniProtKB entry \n",
      "P59594). Proteolytic cleavage sites are marked with arrows. FP , fusion peptide; HR, heptad repeats; TMD, transmembrane domain. (B ) Overview of a multiple \n",
      "sequence alignment of immunodominant peptides with the full-length protein sequences of various alpha- and beta-CoVs (top). Row labels indicate the UniProtKB \n",
      "sequence identifier, start and end positions of enriched peptides (in parentheses), names of the organisms, and cluster numbers (in square brackets). Peptides \n",
      "for which differential enrichment between children and adults were statistically significant (P  ≤ 0.005, Fisher’s exact test) and ORs were at least 2 are indicated \n",
      "with an asterisk. Colors indicate protein domains as shown in A , as well as predicted (pink) (42) and experimentally validated (purple) (56) linear SARS-CoV-2 B cell \n",
      "epitopes. Vertical dashed lines indicate boundaries of regions shown in C –E. The line plot (bottom) shows the mean BepiPred score (blue line) and SD (shaded) for \n",
      "the prediction of linear B cell epitopes among endemic HCoVs. The significance threshold of 0.55 has been marked with a dashed red line. (C –E) Selected regions of page 6 of 15\n",
      "7\n",
      "RESEARCH ARTICLE JCI Insight 2021;6(4):e144499  https://doi.org/10.1172/jci.insight.144499regions of the S protein also appeared more restricted (Figure 5A). Antibody specificities targeting the N pro-\n",
      "tein also showed considerable cross-reactivity with peptides from MERS-CoV , SARS-CoV , and nonhuman \n",
      "CoV isolates. However, the latter cross-reactive antibodies mainly targeted regions rich in serine and arginine \n",
      "residues, with low-complexity sequences and very limited structural conservation, particularly an IDR at the \n",
      "N terminus of the N protein (Figure 5B and Supplemental Figure 9) that lacks a tertiary structure (41). We \n",
      "also detected cross-reactive antibodies targeting the serine- and arginine-rich linker region of the N protein; \n",
      "however, cross-reactivity to this region was largely restricted to peptides derived from nonhuman CoV isolates \n",
      "of domestic animals (Figure 5B), which are more closely related to HCoV-OC43 (6).\n",
      "Finally, we assessed plasma samples of a previously healthy 49-year-old female adult patient who suffered \n",
      "from severe ARDS from SARS-CoV-2 infection, requiring prolonged hospitalization and intensive care. A first \n",
      "sample was obtained 25 days after onset of symptoms and 18 days after intensive care unit admission. A second \n",
      "sample was obtained 1 month later, 1 week after discharge. We compared the antibody profiles at both time \n",
      "points with samples obtained from uninfected family members of the patient, as well as an age- and sex-matched \n",
      "unrelated control. In agreement with what we found in some subjects with a history of endemic HCoV infec-\n",
      "tion, the patient with severe COVID-19 had detectable antibodies that cross-reacted with peptides from SARS-\n",
      "CoV and nonhuman CoVs encoding the furin-like S2′  cleavage site and heptad repeat 2 region of the S protein, \n",
      "peptides from the C-terminal region of the HCoV-HKU1 N protein downstream of the dimerization domain, \n",
      "as well as 2 antigenic sites of the MERS-CoV pp1a (Supplemental Figure 8). We confirmed these findings by \n",
      "analyzing additional plasma samples obtained at a single time point from 6 additional male COVID-19 patients \n",
      "between 5 and 12 days after onset of symptoms (Supplemental Table 7). Indeed, several of the cross-reactive \n",
      "anti-S, anti-N, and anti-pp1a antibodies that we had found in the female COVID-19 patient with severe ARDS \n",
      "were also detectable in male subjects and earlier in the course of infection (Supplemental Table 8).\n",
      "Discussion\n",
      "Our comprehensive and systematic screen for antiviral antibody repertoires across individuals in 3 human \n",
      "cohorts revealed a large number of peptides with novel linear B cell epitopes in several proteins of endemic \n",
      "HCoVs. This is not surprising given that epidemic CoVs, and in particular SARS-CoV , have been the primary \n",
      "focus of previous immunological and epitope screening studies (34, 42). Information about the targets of \n",
      "immune responses to CoVs across different species provides a valuable resource for the prediction of can-\n",
      "didate targets of newly emerging CoVs, as recently shown by Grifoni et al. (42). The authors were able to \n",
      "identify a priori several specific regions of the S, M, and N proteins of SARS-CoV-2 on the basis of sequence \n",
      "homology to the SARS-CoV virus, which are orthologous to several of the immunodominant regions of \n",
      "endemic HCoVs we identified. We detected antibodies against the structural S, N, M, and ORF8 proteins, as \n",
      "well as the nonstructural pp1ab polyprotein of HCoVs, the latter resembling the precursor for the large viral \n",
      "replicase complex (43). Interestingly, in another independent study, Grifoni et al. (28) recently reported sim-\n",
      "ilarly broad T cell responses in COVID-19 patients by employing an analogous screen for T cell epitopes of \n",
      "SARS-CoV-2 proteins using peptide ‘‘megapools’’ in combination with ex vivo T cell assays.\n",
      "Surprisingly, circulating IgG antibodies in children appear to be differentially targeting structural and \n",
      "nonstructural proteins of HCoVs in comparison with adults (Figure 2). Whereas antibody specificities \n",
      "more frequently found in samples of pediatric subjects targeted structural proteins such as the S, N, and \n",
      "M proteins, in adult donors, a region of the nonstructural polyprotein pp1ab containing an acidic tandem \n",
      "repeat (NDDE[D/H]VVTGD) in HCoV-HKU1 appeared to be the predominant target of IgG antibodies. \n",
      "The latter polyprotein is posttranslationally processed into up to 16 subunits that form a large viral replicase \n",
      "complex; however, the function of the acidic tandem repeat and its role in pathogenesis remains unknown \n",
      "(43). This qualitative difference in the antibody repertoires of adult versus pediatric subjects appeared to \n",
      "be a specific characteristic of natural HCoV infection, as we did not find the same degree of variance in \n",
      "the antibody repertoires specific to other common respiratory viruses when comparing our cohorts and \n",
      "different age groups (Supplemental Figure 4). We speculate that the qualitative differences in antibody \n",
      "repertoires of adults versus children reflect a higher frequency and/or more recent exposure of children to \n",
      "seasonal coronaviruses than adults, coupled with the rapid decay of circulating anti-CoV antibodies that the multiple sequence alignment encompassing the N-terminal region of the receptor binding domain (C ), the S1/S2 cleavage site (D ), and the S2′  cleavage site (E ). \n",
      "Amino acid positions on top are shown for UniProtKB entry P59594. Amino acids are marked in color to indicate the level of sequence identity (blue), the proteolytic \n",
      "cleavage sites (black), and linear SARS-CoV-2 B cell epitopes as shown in B . The full sequence alignment is shown in Supplemental Figure 5A.page 7 of 15\n",
      "8\n",
      "RESEARCH ARTICLE JCI Insight 2021;6(4):e144499  https://doi.org/10.1172/jci.insight.144499target the structural proteins of the virions. Further studies will be needed to fully understand the dynamics \n",
      "of antibody responses to endemic HCoVs.\n",
      "Evidence for the transient and dynamic nature of humoral immunity to endemic HCoVs has been provid-\n",
      "ed by numerous human serological studies, although many were conducted with only limited subjects, or only \n",
      "Figure 4. Antigenic regions and predicted antibody binding sites of the N protein. ( A) Schematic representations of the N protein of SARS-CoV (UniProtKB entry \n",
      "P59595). SR-rich, serine- and arginine-rich motif; NLS, predicted nuclear localization sequence; IDR, intrinsically disordered region. (B ) Overview of a multiple \n",
      "sequence alignment of immunodominant peptides with the full-length protein sequences of various alpha- and beta-CoVs (top). Row labels indicate the UniProtKB \n",
      "sequence identifier, start and end positions of enriched peptides (in parentheses), names of the organisms, and cluster numbers (in square brackets). Peptides for \n",
      "which differential enrichment between children and adults was statistically significant (P  ≤ 0.005, Fisher’s exact test) and ORs at least ≥  2 are indicated with an \n",
      "asterisk. Colors indicate protein domains as shown in A  and predicted linear SARS-CoV-2 B cell epitopes (pink) (42). Vertical dashed lines indicate boundaries of \n",
      "the regions shown in C  and D . The line plot (bottom) shows the mean BepiPred score (blue line) and SD (shaded) for the prediction of linear B cell epitopes among \n",
      "endemic HCoVs. The significance threshold of 0.55 has been marked with a dashed red line. (C  and D ) Selected regions of the multiple sequence alignment encom-\n",
      "passing the N-terminal RNA binding domain (C ) and C-terminal self-assembly domain (D ). Amino acid positions on top are shown for UniProtKB reference sequence \n",
      "entry P59595. Amino acids are marked in color to indicate the level of sequence identity (blue) and linear SARS-CoV-2 B cell epitopes (red). The full multiple \n",
      "sequence alignment is shown in Supplemental Figure 5B.page 8 of 15\n",
      "9\n",
      "RESEARCH ARTICLE JCI Insight 2021;6(4):e144499  https://doi.org/10.1172/jci.insight.144499for selected species, and a variety of antibody detection methods were used that are not readily comparable \n",
      "(44–51). Nevertheless, evidence suggests that a sizable proportion of children experience primary infection \n",
      "with endemic HCoVs during their first year of life, and nearly all children have encountered at least one of \n",
      "the endemic HCoVs before 2 years of age, indicating that first exposure to endemic HCoVs occurs very early \n",
      "in life, similar to other common respiratory viruses, such as HRSV or HRVs (45, 49, 50). However, reported \n",
      "seroprevalence rates in older children and adults vary greatly depending on a variety of factors, including \n",
      "age and viral species. There is a general trend indicating that humoral immunity from primary infection with \n",
      "endemic HCoVs wanes quickly and that antibodies detected in older children and adults are rather a conse-\n",
      "quence of more recent reinfections (44–48). We estimated the seroprevalence of antibodies against endemic \n",
      "HCoVs to range between approximately 4% and approximately 27% depending on the species and cohort \n",
      "(Figure 1A). Given that endemic HCoV infections are common and usually acquired during early childhood \n",
      "(45, 46, 49, 50), it is likely that not only the adult subjects, but also many (if not all) of the children aged 7 to \n",
      "15 years that were assessed in our study, have already experienced multiple infections with endemic HCoVs \n",
      "in their lifetime. Therefore, our estimated seroprevalence rates likely reflect the complex dynamics between \n",
      "rates of (re-)infection and waning humoral immunity over time. In agreement with this notion, age was neg-\n",
      "atively associated with seroprevalence in our study, suggesting that the duration of immunity in response to \n",
      "natural infection with endemic HCoVs and/or rates of reinfection decrease with increasing age. The dynam-\n",
      "ics of humoral immunity from past CoV infections is best described in studies of MERS and SARS patients. \n",
      "Although limited in sample size, these studies have shown that antibody titers in all previously infected indi-\n",
      "viduals decline relatively quickly to minimally detectable levels over 2 to 3 years and that patients who suffered \n",
      "from more severe disease had higher and longer lasting total binding antibody titers and neutralizing titers \n",
      "(51). There is also evidence that symptomatic COVID-19 patients mount robust antibody responses that wane \n",
      "quickly over 6 months (52). However, the same study suggested that SARS-CoV-2–specific memory B cells \n",
      "may be sustained over a longer period (52). Indeed, most acute virus infections induce some level of protective \n",
      "and long-term immunity, albeit through a variety of mechanisms that are not necessarily the same for each \n",
      "pathogen and may even differ between hosts due to a variety of factors, including simultaneous viral coin-\n",
      "fection (53, 54). Interestingly, a recent study by Weisberg et al. (55) demonstrated distinct antibody responses \n",
      "to SARS-CoV-2 in children and adults, which were found to be independent of the clinical outcomes and \n",
      "severity of infection. Even children with mild disease generated antibody responses against SARS-CoV-2 \n",
      "with reduced breadth and surprisingly also reduced neutralizing activity compared with adults. It is therefore \n",
      "also possible that children experience an altogether distinct course of infection compared with adults, and \n",
      "Figure 5. Network representation of enriched peptides from structural proteins targeted by cross-reactive antibodies. ( A) Network representation of enriched \n",
      "S protein–derived peptides. (B ) Network representation of enriched N protein–derived peptides. Each node represents an enriched peptide and the color indicates \n",
      "the species. Edges indicate at least 7 amino acids linear sequence identity between 2 nodes (i.e., peptides), the estimated size of a linear B cell epitope. Only net -\n",
      "works of peptides derived from at least 2 species are shown. Labels indicate the cluster number to which each peptide has been assigned. Nodes are represented \n",
      "as spheres if the peptide had been frequently enriched. Nodes marked with a black circle indicate peptides for which differential enrichment between children \n",
      "and adults was statistically significant (P  value ≤  0.005, Fisher’s exact test) and ORs were at least 2. SR-rich, serine- and arginine-rich motif; IDR, intrinsically \n",
      "disordered region; asterisk, region between heptad repeat 1 and heptad repeat 2 of the S2 subunit.page 9 of 15\n",
      "10\n",
      "RESEARCH ARTICLE JCI Insight 2021;6(4):e144499  https://doi.org/10.1172/jci.insight.144499consequently differ in their serological responses, perhaps due to differences in expression levels of the viral \n",
      "receptor, angiotensin-converting enzyme 2, in airway epithelial cells (55). This requires further research. We \n",
      "also found a marginal but significant positive association between seroprevalence of endemic HCoVs and \n",
      "male sex (Figure 1B), which is consistent with an earlier report by Gaunt et al. (5).\n",
      "Despite the variable degree of sequence conservation among different CoV species, the results of our \n",
      "systematic antibody screen highlight that the structural proteins of the virions share common antigenic sites. \n",
      "Indeed, several of the immunodominant regions we have identified experimentally in the structural proteins of \n",
      "endemic HCoVs are orthologous to the regions thought to be immunodominant targets for immune responses \n",
      "to SARS-CoV-2 (42) (Supplemental Figure 8), including 2 linear epitopes on the SARS-CoV-2 S protein that \n",
      "elicit potent neutralizing antibodies in COVID-19 patients (56). Importantly, antigenic regions that we found \n",
      "to be immunodominant in our study (i.e., enriched in ≥ 1% of all samples), as well as those corresponding to \n",
      "peptides for which enrichment was strongly and significantly associated with pediatric subjects (OR ≥  2; P ≤ \n",
      "0.005, Fisher’s exact test), mapped to functionally important regions of the structural CoV proteins. These \n",
      "included regions for receptor binding and the proteolytic cleavage sites of the S protein, as well as the N-ter-\n",
      "minal RNA-binding and C-terminal dimerization domains of the N protein, which have been shown to be \n",
      "critical for virus attachment and entry, cell-to-cell fusion, and virus replication (41, 57–61). The region of the \n",
      "S1 subunit responsible for receptor binding differs considerably among CoV species, which utilize different \n",
      "domains and host cell receptors and consequently differ in their tissue tropism (37–40, 46, 62). However, the S2 \n",
      "subunit resembling the fusion machinery is more conserved, both structurally and in amino acid sequence (38, \n",
      "63) (Supplemental Figure 6). Indeed, we identified an immunodominant and highly conserved linear epitope \n",
      "immediately downstream of the furin-like S2′  cleavage site of the S protein (R¯SA[I/L]ED[I/L]LF) that likely \n",
      "resembles the fusion peptide, although its precise location has been disputed (64). The same antigenic site has \n",
      "recently been found on the SARS-CoV-2 S protein to elicit neutralizing antibodies in COVID-19 patients (56). \n",
      "The high degree of amino acid sequence and conformational conservation of the α -helical region immediately \n",
      "adjacent to the S2′  cleavage site (Figure 3E and Supplemental Figure 6A) likely explains why antibodies tar-\n",
      "geting this region also cross-reacted with orthologous peptides of related CoVs in our study, including those of \n",
      "MERS-CoV , SARS-CoV , and nonhuman isolates, further supporting our overall hypothesis and the important \n",
      "role of this particular region as a pan-CoV target site (40, 56). It is therefore tempting to speculate that at least \n",
      "in some individuals, past infections with endemic HCoVs have elicited cross-reactive antibodies and/or led to \n",
      "the generation of longer lived memory B cells with specific reactivity to this linear epitope, which may provide \n",
      "cross-protection against MERS or COVID-19. This may be the case particularly among children, who are gen-\n",
      "erally less likely to experience severe disease outcomes from infection with epidemic CoVs (17–19, 21). In this \n",
      "context, it is important to highlight that antiviral antibodies can have a variety of protective effector functions \n",
      "that operate through different mechanisms. The underlying mechanisms for antibody-dependent neutralization \n",
      "of enveloped viruses (i.e., the inhibition of virus replication by blocking viral entry into the host cell) include \n",
      "the competitive binding of high-affinity antibodies — via their variable fragment antigen-binding regions — \n",
      "to specific regions within the viral attachment and fusion protein(s) that are also critical for the interaction \n",
      "with the host cell receptor(s) or activating host proteases (56). Neutralizing antibodies may also interfere with \n",
      "the fusion machinery, which undergoes profound activating conformational changes upon viral attachment to \n",
      "overcome the repulsive force between the viral envelope and host cell membrane bilayers (38). Additional anti-\n",
      "body effector functions are Fc mediated, require the participation of additional host immune components, and \n",
      "are not necessarily restricted to antibodies targeting the viral attachment and fusion protein(s). This includes \n",
      "complement-dependent cytotoxicity, as well as enhancement of antibody-dependent cell-mediated cytotoxicity \n",
      "and/or phagocytosis (65, 66). The specific effector functions and mechanisms that are primarily responsible for \n",
      "the generally milder clinical outcome among children when infected with epidemic CoVs remain elusive (55).\n",
      "Broadly cross-reactive antibody responses are also known for other enveloped RNA viruses, which may \n",
      "positively or negatively affect subsequent infection or vaccination. Flaviviruses, for example, are antigenically \n",
      "related, and broadly flavivirus cross-reactive antibodies from previous yellow fever vaccination have been shown \n",
      "to impair and modulate immune responses to tick-borne encephalitis vaccination (67). Similarly, immune histo-\n",
      "ry has been shown to profoundly affect protective B cell responses to influenza (68). Since we detected pan-CoV \n",
      "cross-reactive antibodies less frequently in plasma samples from adult donors, our results argue against a strong \n",
      "therapeutic benefit of intravenous immunoglobulin products to control the spread of COVID-19 disease (69). In \n",
      "this context, it should be noted that large-scale antibody screening by PhIP-Seq may frequently fail to detect con-\n",
      "formational and posttranslationally modified B cell epitopes (29). Nonetheless, we found anti-CoV antibodies  page 10 of 15\n",
      "11\n",
      "RESEARCH ARTICLE JCI Insight 2021;6(4):e144499  https://doi.org/10.1172/jci.insight.144499in plasma of a COVID-19 patient after prolonged hospitalization and intensive care that targeted largely the \n",
      "same structurally conserved and functionally important regions of the viral N and S proteins (Supplemental \n",
      "Figure 8) as those that we detected in a sizable proportion of children, including antibodies binding to the highly \n",
      "conserved motif and furin-like S2′  cleavage site (R¯SA[I/L]ED[I/L]LF), which provides further evidence for the \n",
      "clinical benefit of using convalescent plasma for the prevention and treatment of COVID-19 (65, 70, 71).\n",
      "Our findings may also have important implications for the development of prophylactic or therapeutic \n",
      "monoclonal antibodies and vaccine design, such as in the context of COVID-19 (42, 72). The design of \n",
      "immunogens for next-generation vaccines and the development of monoclonal antibody therapies requires a \n",
      "detailed understanding of the immunogen structure and antibody recognition sites. Endemic HCoVs share \n",
      "common features with epidemic human-tropic CoVs and other enveloped, human-pathogenic viruses, many \n",
      "of which remain obscured by current amino acid sequence alignment tools due to the rapid evolution of \n",
      "viruses. The attachment and fusion protein(s) of enveloped viruses, for example, are key immunogens that \n",
      "share common structural features and employ a similar mechanism for catalyzing membrane fusion between \n",
      "the viral envelope and host cell. The coronavirus S protein belongs to the so-called class I viral fusion proteins, \n",
      "along with the influenza hemagglutinin protein, the HRSV fusion (F) protein, the Ebola virus glycoprotein, \n",
      "and the HIV-1 envelope (Env) protein. An important characteristic of these proteins is their conformational \n",
      "dynamics, which is critical for their function, but this has also proven a major challenge for structural analy-\n",
      "ses. Studies of HIV Env , HRSV F , and the F proteins of other enveloped viruses have highlighted that potent \n",
      "neutralizing antibodies primarily recognize the protein’s prefusion form in the closed conformation and that it \n",
      "is important to stabilize this form for structural analysis (in some cases in complex with bound antibodies), as \n",
      "well as for immunogen design to avoid undesirable antibody responses (38).\n",
      "Methods\n",
      "Study design and samples. We performed a retrospective analysis of deidentified or coded plasma and serum sam-\n",
      "ples collected from 3 human cohorts, namely: (a) 400 healthy male ABD of a blood bank in Qatar with diverse \n",
      "ethnic backgrounds and nationalities (Supplemental Figure 1A); (b) 800 adult male and female Qatari nationals \n",
      "and long-term residents of Qatar who are participating in a national cohort study — the QBB — and who rep-\n",
      "resent the local population in the State of Qatar (31); and (c) 231 pediatric subjects with Qatari nationality who \n",
      "were admitted to, or visited outpatient clinics of, Sidra Medicine. Leftover plasma samples from healthy blood \n",
      "bank donors were collected from 2012 to 2016, deidentified, and stored at –80°C. For this study, specimens from \n",
      "male Qatari nationals 19 to 66 years of age (Supplemental Table 1) were selected from a larger blood donor \n",
      "cohort including 5983 individuals, and then age-matched male donors with other nationalities were randomly \n",
      "selected (Supplemental Figure 1). Samples from female blood bank donors were excluded because they were \n",
      "largely underrepresented in the blood bank donor cohort. We also excluded samples for which age, sex, or \n",
      "nationality information was lacking. Serum samples from the QBB cohort were collected from 2012 to 2017 \n",
      "and were randomly selected samples from the first 3000 individuals taking part in a longitudinal cohort study as \n",
      "described previously (31). Plasma samples from pediatric patients were selected from leftovers of samples pro-\n",
      "cessed in the clinical chemistry labs of Sidra Medicine, a tertiary care hospital for children and women in Doha, \n",
      "Qatar, from September to November 2019. In order to select appropriate pediatric samples, electronic medical \n",
      "records were queried using Discern analytics to identify blood samples from Qatari nationals aged 7 to 15 years \n",
      "for whom basic metabolic panel and comprehensive metabolic panel testing were done in the previous week. \n",
      "Samples from oncology patients, patients requiring complex care, and those in intensive care units, as well as \n",
      "samples from patients with chronic diseases, samples with no centile data, and samples from patients who were \n",
      "underweight (centile < 5%) or overweight were excluded. However, we included obese patients in our analysis, \n",
      "since a considerable proportion of Qatari nationals are overweight. One of the COVID-19 patients assessed in \n",
      "this study was a previously healthy female Belgian national with autosomal recessive interferon regulatory factor \n",
      "7 deficiency who developed ARDS following SARS-CoV-2 infection at the age of 49 (73). For comparison, \n",
      "we also assessed unexposed family members, including the father, mother, brother (heterozygous carriers), and \n",
      "WT sister, as well as an unrelated age- and sex-matched healthy control. Additional male COVID-19 patients \n",
      "assessed in this study were between 30 and 68 years of age and residents of the State of Qatar with diverse \n",
      "nationalities (Supplemental Table 7). All patients assessed here required intensive care for COVID-19; however, \n",
      "information about preexisting comorbidities among the latter patients was not obtained.\n",
      "Phage immunoprecipitation-sequencing. The VirScan phage library used for PhIP-Seq in the present study had \n",
      "been obtained from Stephen Elledge (Brigham and Women’s Hospital and Harvard Medical School, Boston, page 11 of 15\n",
      "12\n",
      "RESEARCH ARTICLE JCI Insight 2021;6(4):e144499  https://doi.org/10.1172/jci.insight.144499Massachusetts, USA). Large-scale serological profiling of the antiviral IgG repertoires in the individual serum \n",
      "or plasma samples was performed as described by Xu et al. (30). Each serum or plasma sample was tested in \n",
      "duplicate, and samples were analyzed in batches with up to 96 samples per batch. Only samples that satisfied \n",
      "a minimum read count of 1 × 106 as well as a Pearson correlation coefficient of at least 0.7 in the 2 technical \n",
      "repeats were considered for downstream analysis. Data from 30 individuals of the ABD cohort and 2 individ-\n",
      "uals of the QBB cohort were excluded from the downstream analysis because of insufficient sequencing read \n",
      "depth, of low sequencing data quality, or 1 of 2 technical replicates had failed (data not shown).\n",
      "Peptide enrichment analysis. To filter for enriched peptides, we first computed −log10P values as described \n",
      "previously (30, 32, 35) by fitting a 0-inflated generalized Poisson model to the distribution of output counts \n",
      "and regressed the parameters for each peptide sequence based on the input read count. We considered a \n",
      "peptide enriched if it passed a reproducibility threshold of 2.3 (−log10P) in 2 technical sample replicates. \n",
      "To remove sporadic hits, we then filtered for antibody specificities to CoV peptides that were found to be \n",
      "enriched in at least 3 of all 1399 subjects assayed and analyzed in this study. We computed species-specific \n",
      "significance cutoff values to estimate the minimum number of enriched, nonhomologous peptides required \n",
      "to consider a sample seropositive using a generalized linear model and in-house serological (ELISA) data \n",
      "from pooled samples that were tested positive for various viruses. We then computed virus score values as \n",
      "described by Xu et al. (30) by counting enriched, nonhomologous peptides for a given species and then \n",
      "adjusted these score values by dividing them with the estimated score cutoff. For this study, we considered a \n",
      "peptide immunodominant if it was enriched in at least 1% of the samples obtained from the 3 larger cohorts \n",
      "(n = 1399) assayed and analyzed in this study.\n",
      "Association studies and differential enrichment analysis. We applied a generalized linear model to test for \n",
      "associations between the HCoV species-specific adjusted score values, sex, and age. We considered an asso-\n",
      "ciation significant if the P value was less than or equal to 0.001. We examined the frequency distribution of \n",
      "enriched peptides among samples of the different age groups (PED versus ABD + QBB) by estimating OR \n",
      "(reported as lod) and associated P values using Fisher’s exact test. Peptides that satisfied both significance \n",
      "(P ≤ 0.005) and magnitude criteria (|OR| ≥ 2) were considered differentially enriched. Positive OR and lod \n",
      "values indicated more frequent peptide enrichment among pediatric study subjects, whereas negative OR \n",
      "and lod values indicated more frequent peptide enrichment among adult subjects.\n",
      "Clustering of  peptides for shared linear B cell epitopes. To estimate the minimum number of linear B cell epi-\n",
      "topes among the enriched peptides, we built a pairwise distance matrix that captured the maximum size of \n",
      "linear sequence identity of amino acids (di,j) between all enriched peptides. Groups of peptides that shared \n",
      "≥ 7 amino acids linear sequence identity (di,j ≥ 7) were assigned to a cluster. Peptides of a given cluster were \n",
      "considered to share a linear B cell epitope (Supplemental Figure 9).\n",
      "Software. For statistical analyses and principal component analysis, we used open-source Python modules \n",
      "detailed below . Multiple sequence alignments were done using the MAFFT (74, 75) via EMBL-EBI’s web ser-\n",
      "vices and Java Alignment Viewer (Jalview) for visualization (76). Network analysis of peptide clusters (Figure 5 \n",
      "and Supplemental Figure 9) was performed using Python module NetworkX (version 2.5). Linear B cell epitopes \n",
      "were predicted using BepiPred-2.0 (77). Protein structures graphics were generated using PyMOL (Schrödinger).\n",
      "Data and code availability. All data are available in the manuscript or the supplemental materials. Raw \n",
      "PhIP-Seq reads and Python in-house scripts used in this study are readily available upon request. The pipeline \n",
      "for processing the PhIP-Seq data has been published previously (29).\n",
      "Statistics. All statistical analyses were performed with Python (version 3.6) using open-source modules, such \n",
      "as SciPy (version 1.14.1), Scikit-learn (version 0.23), Statsmodels (version 0.11), and in-house scripts. Antibody \n",
      "prevalence values for different species (Figure 1A, Supplemental Figure 4A, and Supplemental Table 5) were deter-\n",
      "mined by dividing the number of seropositive individuals of a given group (i.e., cohort, age group, or sex) by the \n",
      "total number of samples of the respective group; calculated values were reported as percentage. For differential \n",
      "enrichment analysis of peptides in pediatric versus adult subjects (Figure 2B, Table 1, and Supplemental Table 6), \n",
      "we used Fisher’s exact test and accounted for multiple testing by Bonferroni’s correction. P  values less than 0.005 \n",
      "were considered statistically significant. For association studies (Figure 1B), we used Student’s t  test (2 tailed) and \n",
      "Bonferroni’s correction and considered associations with P  values less than 0.001 statistically significant.\n",
      "Study approval. The human subject research described here had been approved by the institutional research \n",
      "ethics boards of Sidra Medicine, Qatar Biobank, INSERM, Erasme Hospital, Hamad Medical Corporation, \n",
      "and Qatar University, depending on where subjects were recruited and research was carried out. This includ-\n",
      "ed the receipt of written informed consent from participants.page 12 of 15\n",
      "13\n",
      "RESEARCH ARTICLE JCI Insight 2021;6(4):e144499  https://doi.org/10.1172/jci.insight.144499Author contributions\n",
      "TK and NM conceived the original idea, designed the models and the computational framework of the study, \n",
      "analyzed the data, and wrote manuscript. MR, FA, SSYH, MA, QZ, PB, ZL, EJ, VB, AC, HMY , MKS, LA, \n",
      "and JLC planned or performed the experiments. GKN, AS, IV , JCG, GS, I Migeotte, FH, I Meyts, and MRH \n",
      "contributed samples and data. MR, JLC, and MRH provided critical feedback to this version of the manu-\n",
      "script. All authors have seen and approved the manuscript. It has not been accepted or published elsewhere.\n",
      "Acknowledgments\n",
      "We would like to thank the participating study subjects, the QBB management and staff — in particular \n",
      "Nahla Afifi and Elizabeth Jose, for allowing us to access and analyze samples and data from the QBB — and \n",
      "Stephen Elledge (Brigham and Women’s Hospital and Harvard Medical School) for providing the VirScan \n",
      "phage library used in this study. We would also like to thank Patrick Tang, Mohammed Yousuf Karim, \n",
      "Evonne Chin-Smith, and Damien Chaussabel (Sidra Medicine) for critical reading of the manuscript. This \n",
      "work was supported in part by a grant from the Qatar National Research Fund (PPM1-1220-150017) and \n",
      "funds from Sidra Medicine. I Meyts is a Senior Clinical Investigator at the Research Foundation — Flan-\n",
      "ders and is supported by the CSL Behring Chair of Primary Immunodeficiencies; by the KU Leuven C1 \n",
      "grant C16/18/007; by a VIB GC PID grant; by FWO grants G0C8517N, G0B5120N, and G0E8420N; and \n",
      "by the Jeffrey Modell Foundation. The ULB Center of Human Genetics is supported by the Fonds Erasme.\n",
      "Address correspondence to: Nico Marr, Sidra Medicine, Research Branch, Al Gharrafa Street, Ar-Rayyan, \n",
      "PO Box 26999, Doha, Qatar. Phone: 974.4003.7568; E-mail: nmarr@sidra.org.\n",
      " 1. Hamre D, Procknow JJ. A new virus isolated from the human respiratory tract. Proc Soc Exp Biol Med. 1966;121(1):190–193.\n",
      " 2. McIntosh K, et al. Recovery in tracheal organ cultures of novel viruses from patients with respiratory disease. Proc Natl Acad Sci U S A. \n",
      "1967;57(4):933–940.\n",
      " 3. van der Hoek L, Pyrc K, Jebbink MF , et al. Identification of a new human coronavirus. Nat Med. 2004;10(4):368–373.\n",
      " 4. Woo PCY , et al. Characterization and complete genome sequence of a novel coronavirus, coronavirus HKU1, from patients \n",
      "with pneumonia. J Virol. 2005;79(2):884–895.\n",
      " 5. Gaunt ER, et al. Epidemiology and clinical presentations of the four human coronaviruses 229E, HKU1, NL63, and OC43 \n",
      "detected over 3 years using a novel multiplex real-time PCR method. J Clin Microbiol. 2010;48(8):2940–2947.\n",
      " 6. Corman VM, et al. Hosts and sources of endemic human Coronaviruses. Adv Virus Res. 2018;100:163–188.\n",
      " 7. Kristoffersen A W , et al. Coronavirus causes lower respiratory tract infections less frequently than RSV in hospitalized Norwegian \n",
      "children. Pediatr Infect Dis J. 2011;30(4):279–283.\n",
      " 8. Drosten C, et al. Identification of a novel coronavirus in patients with severe acute respiratory syndrome. N Engl J Med. \n",
      "2003;348(20):1967–1976.\n",
      " 9. Zaki AM, et al. Isolation of a novel coronavirus from a man with pneumonia in Saudi Arabia. N Engl J Med. 2012;367(19):1814–1820.\n",
      " 10. Zhu N, et al. A novel coronavirus from patients with pneumonia in China, 2019.  N Engl J Med. 2020;382(8):727–733.\n",
      " 11. Dong E, et al. An interactive web-based dashboard to track COVID-19 in real time. Lancet Infect Dis. 2020;20(5):533–534.\n",
      " 12. de Wit E, et al. SARS and MERS: recent insights into emerging coronaviruses. Nat Rev Microbiol. 2016;14(8):523–534.\n",
      " 13. CDC COVID-19 Response Team. Severe outcomes among patients with coronavirus disease 2019 (COVID-19) — United States, \n",
      "February 12–March 16, 2020. MMWR Morb Mortal Wkly Rep. 2020;69(12):343–346.\n",
      " 14. Li X, Ma X. Acute respiratory failure in COVID-19: is it “typical” ARDS? Crit Care. 2020;24(1):198.\n",
      " 15. Vijgen L, et al. Evolutionary history of the closely related group 2 coronaviruses: porcine hemagglutinating encephalomyelitis \n",
      "virus, bovine coronavirus, and human coronavirus OC43. J Virol. 2006;80(14):7270–7274.\n",
      " 16. Young BE, et al. Epidemiologic features and clinical course of patients infected with SARS-CoV-2 in Singapore. JAMA. \n",
      "2020;323(15):1488–1494.\n",
      " 17. Dong Y , et al. Epidemiology of COVID-19 among children in China. Pediatrics. 2020;145(6):20200702.\n",
      " 18. Zimmermann P , Curtis N. Coronavirus infections in children including COVID-19: an overview of the epidemiology, clinical features, \n",
      "diagnosis, treatment and prevention options in children. Pediatr Infect Dis J. 2020;39(5):355–368.\n",
      " 19. Al-Tawfiq JA, et al. Middle East respiratory syndrome coronavirus disease is rare in children: an update from Saudi Arabia. \n",
      "World J Clin Pediatr. 2016;5(4):391–396.\n",
      " 20. Zhang J, et al. Changes in contact patterns shape the dynamics of the COVID-19 outbreak in China. Science. \n",
      "2020;368(6498):1481–1486.\n",
      " 21. Davies NG, et al. Age-dependent effects in the transmission and control of COVID-19 epidemics. Nat Med. 2020;26(8):1205–1211.\n",
      " 22. Guo L, et al. Profiling early humoral response to diagnose novel coronavirus disease (COVID-19). Clin Infect Dis. 2020;71(15):778–785.\n",
      " 23. Okba NMA, et al. Severe acute respiratory syndrome coronavirus 2-specific antibody responses in coronavirus disease patients. \n",
      "Emerg Infect Dis. 2020;26(7):1478–1488.\n",
      " 24. Yongchen Z, et al. Different longitudinal patterns of nucleic acid and serology testing results based on disease severity of \n",
      "COVID-19 patients. Emerg Microbes Infect. 2020;9(1):833–836.\n",
      " 25. Khan S, et al. Analysis of serologic cross-reactivity between common human coronaviruses and SARS-CoV-2 using coronavirus \n",
      "antigen microarray [preprint]. https://doi.org/10.1101/2020.03.24.006544. Posted on bioRxiv March 25, 2020.page 13 of 15\n",
      "14\n",
      "RESEARCH ARTICLE JCI Insight 2021;6(4):e144499  https://doi.org/10.1172/jci.insight.144499 26. Smatti MK, et al. Viral-induced enhanced disease illness. Front Microbiol. 2018;9:2991.\n",
      " 27. Braun J, et al. SARS-CoV-2-reactive T cells in healthy donors and patients with COVID-19. Nature. 2020;587(7833):270–274.\n",
      " 28. Grifoni A, et al. Targets of T cell responses to SARS-CoV-2 Coronavirus in humans with COVID-19 disease and unexposed \n",
      "individuals. Cell. 2020;181(7):1489–1501.\n",
      " 29. Mohan D, et al. Publisher correction: PhIP-Seq characterization of serum antibodies using oligonucleotide-encoded peptidomes. \n",
      "Nat Protoc. 2019;14(8):2596.\n",
      " 30. Xu GJ, et al. Viral immunology. Comprehensive serological profiling of human populations using a synthetic human virome. \n",
      "Science. 2015;348(6239):aaa0698.\n",
      " 31. Al Kuwari H, et al. The Qatar Biobank: background and methods. BMC Public Health. 2015;15:1208.\n",
      " 32. Mina MJ, et al. Measles virus infection diminishes preexisting antibodies that offer protection from other pathogens. Science. \n",
      "2019;366(6465):599–606.\n",
      " 33. Drutman SB, et al. Fatal cytomegalovirus infection in an adult with inherited NOS2 deficiency. N Engl J Med. 2020;382(5):437–445.\n",
      " 34. Vita R, et al. The Immune Epitope Database (IEDB): 2018 update. Nucleic Acids Res. 2019;47(D1):D339–D343.\n",
      " 35. Pou C, et al. The repertoire of maternal anti-viral antibodies in human newborns. Nat Med. 2019;25(4):591–596.\n",
      " 36. Tortorici MA, et al. Structural basis for human coronavirus attachment to sialic acid receptors. Nat Struct Mol Biol. 2019;26(6):481–489.\n",
      " 37. Tortorici MA, Veesler D. Structural insights into coronavirus entry. Adv Virus Res. 2019;105:93–116.\n",
      " 38. Rey FA, Lok SM. Common features of enveloped viruses and implications for immunogen design for next-generation vaccines. \n",
      "Cell. 2018;172(6):1319–1334.\n",
      " 39. Hulswit RJG, et al. Human coronaviruses OC43 and HKU1 bind to 9-O-acetylated sialic acids via a conserved receptor-binding \n",
      "site in spike protein domain A. Proc Natl Acad Sci U S A. 2019;116(7):2681–2690.\n",
      " 40. Walls AC, Pet al. Structure, function, and antigenicity of the SARS-CoV-2 spike glycoprotein. Cell. 2020;181(2):281–292.\n",
      " 41. McBride R, et al. The coronavirus nucleocapsid is a multifunctional protein. Viruses. 2014;6(8):2991–3018.\n",
      " 42. Grifoni A, et al. A sequence homology and bioinformatic approach can predict candidate targets for immune responses to \n",
      "SARS-CoV-2. Cell Host Microbe. 2020;27(4):671–680.\n",
      " 43. Ziebuhr J. The coronavirus replicase. Curr Top Microbiol Immunol. 2005;287:57–94.\n",
      " 44. Monto AS, Lim SK. The Tecumseh study of respiratory illness. VI. Frequency of and relationship between outbreaks of coronavirus \n",
      "infection. J Infect Dis. 1974;129(3):271–276.\n",
      " 45. Shao X, et al. Seroepidemiology of group I human coronaviruses in children. J Clin Virol. 2007;40(3):207–213.\n",
      " 46. Hofmann H, et al. Human coronavirus NL63 employs the severe acute respiratory syndrome coronavirus receptor for cellular entry. \n",
      "Proc Natl Acad Sci U S A. 2005;102(22):7988–7993.\n",
      " 47. Hamre D, Beem M. Virologic studies of acute respiratory disease in young adults. V . Coronavirus 229E infections during six \n",
      "years of surveillance. Am J Epidemiol. 1972;96(2):94–106.\n",
      " 48. Chan CM, et al. Examination of seroprevalence of coronavirus HKU1 infection with S protein-based ELISA and neutralization \n",
      "assay against viral spike pseudotyped virus. J Clin Virol. 2009;45(1):54–60.\n",
      " 49. Dijkman R, et al. The dominance of human coronavirus OC43 and NL63 infections in infants. J Clin Virol. 2012;53(2):135–139.\n",
      " 50. Dijkman R, et al. Human coronavirus NL63 and 229E seroconversion in children. J Clin Microbiol. 2008;46(7):2368–2373.\n",
      " 51. Kellam P , Barclay W . The dynamics of humoral immune responses following SARS-CoV-2 infection and the potential for reinfection. \n",
      "J Gen Virol. 2020;101(8):791–797.\n",
      " 52. Vaisman-Mentesh A, et al. SARS-CoV-2 specific memory B cells frequency in recovered patient remains stable while antibodies \n",
      "decay over time [preprint]. https://doi.org/10.1101/2020.08.23.2017979. Posted on medRxiv August 25, 2020.\n",
      " 53. Sallusto F , et al. From vaccines to memory and back. Immunity. 2010;33(4):451–463.\n",
      " 54. Kenney LL, et al. Increased immune response variability during simultaneous viral coinfection leads to unpredictability in CD8 \n",
      "T cell immunity and pathogenesis. J Virol. 2015;89(21):10786–10801.\n",
      " 55. Weisberg SP , et al. Distinct antibody responses to SARS-CoV-2 in children and adults across the COVID-19 clinical spectrum. \n",
      "Nat Immunol. 2021;22(1):25–31.\n",
      " 56. Poh CM, et al. Two linear epitopes on the SARS-CoV-2 spike protein that elicit neutralising antibodies in COVID-19 patients. \n",
      "Nat Commun. 2020;11(1):2806.\n",
      " 57. Tylor S, et al. The SR-rich motif in SARS-CoV nucleocapsid protein is important for virus replication. Can J Microbiol. \n",
      "2009;55(3):254–260.\n",
      " 58. Surjit M, Lal SK. The SARS-CoV nucleocapsid protein: a protein with multifarious activities. Infect Genet Evol. 2008;8(4):397–405.\n",
      " 59. Li F . Structure, function, and evolution of Coronavirus Spike proteins. Annu Rev Virol. 2016;3(1):237–261.\n",
      " 60. Follis KE, et al. Furin cleavage of the SARS coronavirus spike glycoprotein enhances cell-cell fusion but does not affect virion entry. \n",
      "Virology. 2006;350(2):358–369.\n",
      " 61. Belouzard S, et al. Mechanisms of coronavirus cell entry mediated by the viral spike protein. Viruses. 2012;4(6):1011–1033.\n",
      " 62. Fung TS, Liu DX. Human Coronavirus: host-pathogen interaction. Annu Rev Microbiol. 2019;73:529–557.\n",
      " 63. Walls AC, et al. Unexpected receptor functional mimicry elucidates activation of coronavirus fusion. Cell. 2019;176(5):1026–1039.\n",
      " 64. Lai AL, et al. The SARS-CoV fusion peptide forms an extended bipartite fusion platform that perturbs membrane order in a \n",
      "calcium-dependent manner. J Mol Biol. 2017;429(24):3875–3892.\n",
      " 65. Bloch EM, et al. Deployment of convalescent plasma for the prevention and treatment of COVID-19. J Clin Invest. \n",
      "2020;130(6):2757–2765.\n",
      " 66. van Erp EA, et al. Fc-mediated antibody effector functions during respiratory syncytial virus infection and disease. Front Immunol. \n",
      "2019;10:548.\n",
      " 67. Bradt V , et al. Pre-existing yellow fever immunity impairs and modulates the antibody response to tick-borne encephalitis vaccination. \n",
      "NPJ Vaccines. 2019;4(1):38.\n",
      " 68. Andrews SF , et al. Immune history profoundly affects broadly protective B cell responses to influenza. Sci Transl Med. \n",
      "2015;7(316):316ra192.\n",
      " 69. Diez JM, et al. Currently available intravenous immunoglobulin contains antibodies reacting against severe acute respiratory \n",
      "syndrome coronavirus 2 antigens. Immunotherapy. 2020;12(8):571–576.page 14 of 15\n",
      "15\n",
      "RESEARCH ARTICLE JCI Insight 2021;6(4):e144499  https://doi.org/10.1172/jci.insight.144499 70. Benner SE, et al. SARS-CoV-2 antibody avidity responses in COVID-19 patients and convalescent plasma donors. J Infect Dis. \n",
      "2020;222(12):1974–1984.\n",
      " 71. Casadevall A, Pirofski LA. The convalescent sera option for containing COVID-19. J Clin Invest. 2020;130(4):1545–1548.\n",
      " 72. Zhou G, Zhao Q. Perspectives on therapeutic neutralizing antibodies against the novel Coronavirus SARS-CoV-2. Int J Biol Sci. \n",
      "2020;16(10):1718–1723.\n",
      " 73. Zhang Q, et al. Inborn errors of type I IFN immunity in patients with life-threatening COVID-19. Science. 2020;370(6515):eabd4570.\n",
      " 74. Madeira F , et al. The EMBL-EBI search and sequence analysis tools APIs in 2019. Nucleic Acids Res. 2019;47(W1):W636–W641.\n",
      " 75. Katoh K, Standley DM. MAFFT multiple sequence alignment software version 7: improvements in performance and usability. \n",
      "Mol Biol Evol. 2013;30(4):772–780.\n",
      " 76. Madeira F , et al. Using EMBL-EBI services via web interface and programmatically via web services. Curr Protoc Bioinformatics. \n",
      "2019;66(1):74.\n",
      " 77. Dhanda SK, et al. IEDB-AR: immune epitope database-analysis resource in 2019. Nucleic Acids Res. 2019;47(W1):W502–W506.\n",
      "\n",
      "Generating final response...\n",
      "\n",
      "\n",
      "\u001b[94mThe\u001b[0m\u001b[94m provided\u001b[0m\u001b[94m text\u001b[0m\u001b[94m appears\u001b[0m\u001b[94m to\u001b[0m\u001b[94m be\u001b[0m\u001b[94m a\u001b[0m\u001b[94m list\u001b[0m\u001b[94m of\u001b[0m\u001b[94m references\u001b[0m\u001b[94m related\u001b[0m\u001b[94m to\u001b[0m\u001b[94m the\u001b[0m\u001b[94m COVID\u001b[0m\u001b[94m-\u001b[0m\u001b[94m19\u001b[0m\u001b[94m pandemic\u001b[0m\u001b[94m and\u001b[0m\u001b[94m immunity\u001b[0m\u001b[94m.\u001b[0m\u001b[94m The\u001b[0m\u001b[94m references\u001b[0m\u001b[94m are\u001b[0m\u001b[94m formatted\u001b[0m\u001b[94m according\u001b[0m\u001b[94m to\u001b[0m\u001b[94m the\u001b[0m\u001b[94m style\u001b[0m\u001b[94m typically\u001b[0m\u001b[94m used\u001b[0m\u001b[94m in\u001b[0m\u001b[94m scientific\u001b[0m\u001b[94m publications\u001b[0m\u001b[94m,\u001b[0m\u001b[94m with\u001b[0m\u001b[94m authors\u001b[0m\u001b[94m'\u001b[0m\u001b[94m names\u001b[0m\u001b[94m,\u001b[0m\u001b[94m article\u001b[0m\u001b[94m titles\u001b[0m\u001b[94m,\u001b[0m\u001b[94m journal\u001b[0m\u001b[94m names\u001b[0m\u001b[94m,\u001b[0m\u001b[94m publication\u001b[0m\u001b[94m years\u001b[0m\u001b[94m,\u001b[0m\u001b[94m and\u001b[0m\u001b[94m page\u001b[0m\u001b[94m numbers\u001b[0m\u001b[94m.\n",
      "\n",
      "\u001b[0m\u001b[94mHere\u001b[0m\u001b[94m's\u001b[0m\u001b[94m a\u001b[0m\u001b[94m breakdown\u001b[0m\u001b[94m of\u001b[0m\u001b[94m the\u001b[0m\u001b[94m content\u001b[0m\u001b[94m:\n",
      "\n",
      "\u001b[0m\u001b[94m*\u001b[0m\u001b[94m The\u001b[0m\u001b[94m first\u001b[0m\u001b[94m \u001b[0m\u001b[94m4\u001b[0m\u001b[94m references\u001b[0m\u001b[94m (\u001b[0m\u001b[94m55\u001b[0m\u001b[94m-\u001b[0m\u001b[94m58\u001b[0m\u001b[94m)\u001b[0m\u001b[94m discuss\u001b[0m\u001b[94m the\u001b[0m\u001b[94m S\u001b[0m\u001b[94mARS\u001b[0m\u001b[94m-Co\u001b[0m\u001b[94mV\u001b[0m\u001b[94m nucle\u001b[0m\u001b[94moc\u001b[0m\u001b[94maps\u001b[0m\u001b[94mid\u001b[0m\u001b[94m protein\u001b[0m\u001b[94m and\u001b[0m\u001b[94m its\u001b[0m\u001b[94m importance\u001b[0m\u001b[94m for\u001b[0m\u001b[94m virus\u001b[0m\u001b[94m replication\u001b[0m\u001b[94m.\n",
      "\u001b[0m\u001b[94m*\u001b[0m\u001b[94m References\u001b[0m\u001b[94m \u001b[0m\u001b[94m59\u001b[0m\u001b[94m-\u001b[0m\u001b[94m63\u001b[0m\u001b[94m provide\u001b[0m\u001b[94m information\u001b[0m\u001b[94m on\u001b[0m\u001b[94m the\u001b[0m\u001b[94m coronavirus\u001b[0m\u001b[94m spike\u001b[0m\u001b[94m protein\u001b[0m\u001b[94m and\u001b[0m\u001b[94m its\u001b[0m\u001b[94m role\u001b[0m\u001b[94m in\u001b[0m\u001b[94m viral\u001b[0m\u001b[94m entry\u001b[0m\u001b[94m and\u001b[0m\u001b[94m fusion\u001b[0m\u001b[94m with\u001b[0m\u001b[94m host\u001b[0m\u001b[94m cells\u001b[0m\u001b[94m.\n",
      "\u001b[0m\u001b[94m*\u001b[0m\u001b[94m References\u001b[0m\u001b[94m \u001b[0m\u001b[94m64\u001b[0m\u001b[94m-\u001b[0m\u001b[94m66\u001b[0m\u001b[94m focus\u001b[0m\u001b[94m on\u001b[0m\u001b[94m conv\u001b[0m\u001b[94males\u001b[0m\u001b[94mcent\u001b[0m\u001b[94m plasma\u001b[0m\u001b[94m,\u001b[0m\u001b[94m which\u001b[0m\u001b[94m is\u001b[0m\u001b[94m a\u001b[0m\u001b[94m treatment\u001b[0m\u001b[94m that\u001b[0m\u001b[94m uses\u001b[0m\u001b[94m antibodies\u001b[0m\u001b[94m from\u001b[0m\u001b[94m recovered\u001b[0m\u001b[94m COVID\u001b[0m\u001b[94m-\u001b[0m\u001b[94m19\u001b[0m\u001b[94m patients\u001b[0m\u001b[94m to\u001b[0m\u001b[94m help\u001b[0m\u001b[94m treat\u001b[0m\u001b[94m others\u001b[0m\u001b[94m.\n",
      "\u001b[0m\u001b[94m*\u001b[0m\u001b[94m References\u001b[0m\u001b[94m \u001b[0m\u001b[94m67\u001b[0m\u001b[94m-\u001b[0m\u001b[94m69\u001b[0m\u001b[94m explore\u001b[0m\u001b[94m how\u001b[0m\u001b[94m pre\u001b[0m\u001b[94m-existing\u001b[0m\u001b[94m immunity\u001b[0m\u001b[94m can\u001b[0m\u001b[94m affect\u001b[0m\u001b[94m antibody\u001b[0m\u001b[94m responses\u001b[0m\u001b[94m to\u001b[0m\u001b[94m vaccination\u001b[0m\u001b[94m or\u001b[0m\u001b[94m infection\u001b[0m\u001b[94m.\n",
      "\u001b[0m\u001b[94m*\u001b[0m\u001b[94m References\u001b[0m\u001b[94m \u001b[0m\u001b[94m70\u001b[0m\u001b[94m-\u001b[0m\u001b[94m72\u001b[0m\u001b[94m discuss\u001b[0m\u001b[94m the\u001b[0m\u001b[94m use\u001b[0m\u001b[94m of\u001b[0m\u001b[94m therapeutic\u001b[0m\u001b[94m neutral\u001b[0m\u001b[94mizing\u001b[0m\u001b[94m antibodies\u001b[0m\u001b[94m against\u001b[0m\u001b[94m S\u001b[0m\u001b[94mARS\u001b[0m\u001b[94m-Co\u001b[0m\u001b[94mV\u001b[0m\u001b[94m-\u001b[0m\u001b[94m2\u001b[0m\u001b[94m.\n",
      "\u001b[0m\u001b[94m*\u001b[0m\u001b[94m Reference\u001b[0m\u001b[94m \u001b[0m\u001b[94m73\u001b[0m\u001b[94m reports\u001b[0m\u001b[94m on\u001b[0m\u001b[94m in\u001b[0m\u001b[94mborn\u001b[0m\u001b[94m errors\u001b[0m\u001b[94m of\u001b[0m\u001b[94m type\u001b[0m\u001b[94m I\u001b[0m\u001b[94m IF\u001b[0m\u001b[94mN\u001b[0m\u001b[94m immunity\u001b[0m\u001b[94m in\u001b[0m\u001b[94m patients\u001b[0m\u001b[94m with\u001b[0m\u001b[94m severe\u001b[0m\u001b[94m COVID\u001b[0m\u001b[94m-\u001b[0m\u001b[94m19\u001b[0m\u001b[94m.\n",
      "\u001b[0m\u001b[94m*\u001b[0m\u001b[94m References\u001b[0m\u001b[94m \u001b[0m\u001b[94m74\u001b[0m\u001b[94m-\u001b[0m\u001b[94m77\u001b[0m\u001b[94m describe\u001b[0m\u001b[94m various\u001b[0m\u001b[94m bio\u001b[0m\u001b[94minformatics\u001b[0m\u001b[94m tools\u001b[0m\u001b[94m and\u001b[0m\u001b[94m resources\u001b[0m\u001b[94m used\u001b[0m\u001b[94m in\u001b[0m\u001b[94m the\u001b[0m\u001b[94m analysis\u001b[0m\u001b[94m,\u001b[0m\u001b[94m including\u001b[0m\u001b[94m sequence\u001b[0m\u001b[94m alignment\u001b[0m\u001b[94m software\u001b[0m\u001b[94m (\u001b[0m\u001b[94mMA\u001b[0m\u001b[94mFFT\u001b[0m\u001b[94m),\u001b[0m\u001b[94m database\u001b[0m\u001b[94m analysis\u001b[0m\u001b[94m (\u001b[0m\u001b[94mIED\u001b[0m\u001b[94mB\u001b[0m\u001b[94m-\u001b[0m\u001b[94mAR\u001b[0m\u001b[94m),\u001b[0m\u001b[94m and\u001b[0m\u001b[94m search\u001b[0m\u001b[94m services\u001b[0m\u001b[94m (\u001b[0m\u001b[94mEM\u001b[0m\u001b[94mBL\u001b[0m\u001b[94m-\u001b[0m\u001b[94mEB\u001b[0m\u001b[94mI\u001b[0m\u001b[94m).\n",
      "\n",
      "\u001b[0m\u001b[94mThe\u001b[0m\u001b[94m text\u001b[0m\u001b[94m does\u001b[0m\u001b[94m not\u001b[0m\u001b[94m provide\u001b[0m\u001b[94m a\u001b[0m\u001b[94m detailed\u001b[0m\u001b[94m summary\u001b[0m\u001b[94m or\u001b[0m\u001b[94m conclusion\u001b[0m\u001b[94m but\u001b[0m\u001b[94m rather\u001b[0m\u001b[94m serves\u001b[0m\u001b[94m as\u001b[0m\u001b[94m a\u001b[0m\u001b[94m reference\u001b[0m\u001b[94m list\u001b[0m\u001b[94m for\u001b[0m\u001b[94m further\u001b[0m\u001b[94m research\u001b[0m\u001b[94m on\u001b[0m\u001b[94m COVID\u001b[0m\u001b[94m-\u001b[0m\u001b[94m19\u001b[0m\u001b[94m immunity\u001b[0m\u001b[94m.\u001b[0m\u001b[94m\u001b[0m"
     ]
    }
   ],
   "source": [
    "#| hide\n",
    "assistant.chat(\"Do you have access to the abstract?\", show_progress=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Selecting tools...\n",
      "\n",
      "[{'function': {'name': 'query_openalex_api', 'arguments': {'identifyer': 'article DOI or title', 'identifyer_type': 'DOI'}}}]\n",
      "Calling query_openalex_api() with arguments {'identifyer': 'article DOI or title', 'identifyer_type': 'DOI'}...\n",
      "\n"
     ]
    },
    {
     "ename": "UnboundLocalError",
     "evalue": "cannot access local variable 'filter' where it is not associated with a value",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mUnboundLocalError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[19], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;66;03m#| hide\u001b[39;00m\n\u001b[0;32m----> 2\u001b[0m \u001b[43massistant\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mchat\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mHow often has this article been cited?\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mshow_progress\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/scholaris-env/lib/python3.12/site-packages/scholaris/core.py:995\u001b[0m, in \u001b[0;36mAssistant.chat\u001b[0;34m(self, prompt, show_progress)\u001b[0m\n\u001b[1;32m    993\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m show_progress: \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mCalling \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mtool[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mfunction\u001b[39m\u001b[38;5;124m'\u001b[39m][\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mname\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m() with arguments \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mtool[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mfunction\u001b[39m\u001b[38;5;124m'\u001b[39m][\u001b[38;5;124m'\u001b[39m\u001b[38;5;124marguments\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m...\u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m    994\u001b[0m args \u001b[38;5;241m=\u001b[39m tool[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mfunction\u001b[39m\u001b[38;5;124m'\u001b[39m][\u001b[38;5;124m'\u001b[39m\u001b[38;5;124marguments\u001b[39m\u001b[38;5;124m'\u001b[39m]\n\u001b[0;32m--> 995\u001b[0m function_response \u001b[38;5;241m=\u001b[39m \u001b[43mfunction_to_call\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    996\u001b[0m \u001b[38;5;66;03m# print(f\"Function response type: {type(function_response)}\\n\") # Uncomment for debugging\u001b[39;00m\n\u001b[1;32m    997\u001b[0m \u001b[38;5;66;03m# print(f\"Function response: {function_response}\\n\") # Uncomment for debugging\u001b[39;00m\n\u001b[1;32m    999\u001b[0m function_response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mstr\u001b[39m(function_response) \u001b[38;5;28;01mif\u001b[39;00m function_response \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m\n",
      "File \u001b[0;32m~/miniconda3/envs/scholaris-env/lib/python3.12/site-packages/scholaris/core.py:120\u001b[0m, in \u001b[0;36mjson_schema_decorator.<locals>.wrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    118\u001b[0m \u001b[38;5;129m@functools\u001b[39m\u001b[38;5;241m.\u001b[39mwraps(func)\n\u001b[1;32m    119\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mwrapper\u001b[39m(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[0;32m--> 120\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mfunc\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/scholaris-env/lib/python3.12/site-packages/scholaris/core.py:668\u001b[0m, in \u001b[0;36mquery_openalex_api\u001b[0;34m(identifyer, identifyer_type)\u001b[0m\n\u001b[1;32m    663\u001b[0m         url \u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mbase_url\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124mworks/https://doi.org/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00midentifyer\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    664\u001b[0m         \u001b[38;5;28mfilter\u001b[39m \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    666\u001b[0m params \u001b[38;5;241m=\u001b[39m {\n\u001b[1;32m    667\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmailto\u001b[39m\u001b[38;5;124m\"\u001b[39m: EMAIL,\n\u001b[0;32m--> 668\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mfilter\u001b[39m\u001b[38;5;124m\"\u001b[39m: \u001b[38;5;28;43mfilter\u001b[39;49m,\n\u001b[1;32m    669\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mpage\u001b[39m\u001b[38;5;124m\"\u001b[39m: \u001b[38;5;241m1\u001b[39m,\n\u001b[1;32m    670\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mper-page\u001b[39m\u001b[38;5;124m\"\u001b[39m: \u001b[38;5;241m5\u001b[39m,\n\u001b[1;32m    671\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mselect\u001b[39m\u001b[38;5;124m\"\u001b[39m: \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mid,doi,title,publication_year,cited_by_count,cited_by_api_url,open_access,type,type_crossref\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m    672\u001b[0m }\n\u001b[1;32m    674\u001b[0m response \u001b[38;5;241m=\u001b[39m requests\u001b[38;5;241m.\u001b[39mget(url, params\u001b[38;5;241m=\u001b[39mparams)\n\u001b[1;32m    676\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m response\u001b[38;5;241m.\u001b[39mstatus_code \u001b[38;5;241m!=\u001b[39m \u001b[38;5;241m200\u001b[39m:\n",
      "\u001b[0;31mUnboundLocalError\u001b[0m: cannot access local variable 'filter' where it is not associated with a value"
     ]
    }
   ],
   "source": [
    "#| hide\n",
    "assistant.chat(\"How often has this article been cited?\", show_progress=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#| default_exp demo"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "python3",
   "language": "python",
   "name": "python3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
